langkit/topics.py

```diff
@@ -12,16 +13,23 @@
 classifier = pipeline("zero-shot-classification", model=model_path)
 
 
-@register_metric_udf(col_type=String)
-def closest_topic(text: str) -> str:
-    output = classifier(text, _topics, multi_label=False)
-    return output["labels"][0]
+def closest_topic(text):
+    index = text.columns[0] if isinstance(text, pd.DataFrame) else list(text.keys())[0]
+    result = []
+    for input in text[index]:
+        output = classifier(input, _topics, multi_label=False)
+        result.append(output["labels"][0])
+    return result
 
 
-def init(topics: Optional[list] = None):
+def init(topics: Optional[List] = None):
```

Comment:
Consider: It might be worth adding a list of columns to register as an optional param to init here. If None we could default to using the lang_config values you have below.

Reply:
Perhaps do that in a separate PR